### PR TITLE
Add snaps name provider

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## **Description**

Add the `SnapsNameProvider` to use installed snaps with the `name-lookup` endowment to propose names for Ethereum addresses in the `Name` component.

## **Manual Testing Steps**

1. Go to test snaps.
2. Install the `Name Lookup` snap.
3. Go to test dApp.
4. Create a `V3` or `V4` typed signature request.
5. Click an address.
6. Verify the proposed names include a `Name Lookup Example Snap` option.

## **Screenshots**

snaps_provider

## **Related Issues**

Fixes [#1235](https://github.com/MetaMask/MetaMask-planning/issues/1235)

## **Pre-merge Author Checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge Reviewer Checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by vinistevam — DISMISSED** [review]
:rocket:  This PR is brilliant.

**Review by vinistevam — DISMISSED** [review]

**Review by vinistevam — DISMISSED** [review]

File: test/e2e/snaps/petnames.spec.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ const { strict: assert } = require('assert');
 const { withFixtures, openDapp, convertToHexValue } = require('../helpers');
 const FixtureBuilder = require('../fixture-builder');
 
+const TEST_SNAPS_URL = 'https://metamask.github.io/snaps/test-snaps/1.1.0';
```
Comment (FrederikBolding) [inline test/e2e/snaps/petnames.spec.js:5]: Import from https://github.com/MetaMask/metamask-extension/blob/993be35490e5833af6e42ddd87499cae875f39d9/test/e2e/snaps/enums.js#L2?

File: app/scripts/lib/SnapsNameProvider.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    const snaps = this.#getNameLookupSnaps();
+
+    const sourceIds = {
+      [NameType.ETHEREUM_ADDRESS]: snaps.map((snap) => snap.id),
+    };
+
+    const sourceLabels = snaps.reduce(
+      (acc: NameProviderMetadata['sourceLabels'], snap) => {
+        const snapDetails = this.#messenger.call('SnapController:get', snap.id);
+        const snapName = snapDetails?.manifest.proposedName;
```
Comment (FrederikBolding) [inline app/scripts/lib/SnapsNameProvider.ts:55]: May want to consider using `getSnapName`

**Review by FrederikBolding — APPROVED** [review]

File: test/e2e/snaps/petnames.spec.js (hunk 1 above)
Comment (matthewwalsh0) [reply test/e2e/snaps/petnames.spec.js]: Done.

File: app/scripts/lib/SnapsNameProvider.ts (hunk 2 above)
Comment (matthewwalsh0) [reply app/scripts/lib/SnapsNameProvider.ts:55]: Good point, will add a task to our backlog to move `getSnapName` into shared and use it here.

**Review by vinistevam — APPROVED** [review]

**Pull request merged**